src/python/pants/backend/jvm/tasks/check_banned_deps.py

```diff
@@ -0,0 +1,166 @@
+      """
+      Check whether a given target complies with this constraint.
+      :param target: Target to check.
+      :param task_context: Context of the task, to extract from it any relevant information.
+      :param classes_relevant_to_target: Fully qualified names of the classes in the target and its dependencies.
+      :return:
+      """
+      banned_classes = [c for c in classes_relevant_to_target if c.startswith(self.banned_package_name)]
+      if banned_classes:
+        raise DependencyConstraint.BannedDependencyException(
+          'Target {} uses rule "{}" to ban classes ({})'.format(
+            target.target_base,
+            self.banned_package_name,
+            ", ".join(banned_classes)
+          ))
+
+    def _compute_fingerprint(self):
+      return stable_json_hash(self.banned_package_name)
+
+  class Tag(Constraint):
+    """Representation of the ban of a tag"""
+
+    def __init__(self, banned_tag_name):
+      self.banned_tag_name = banned_tag_name
+
+    def _has_bad_tag(self, target):
+      return self.banned_tag_name in target.tags
+
+    def check_target(self, target, task_context, classes_relevant_to_target):
+      relevant_targets = DependencyConstraints.relevant_targets(target)
+      forbidden_targets = [t for t in relevant_targets if self._has_bad_tag(t)]
+      if forbidden_targets:
+        raise DependencyConstraint.BannedDependencyException(
+          'Target {} has baned tag "{}", but these targets have it ({})'.format(
+            target.target_base,
+            self.banned_tag_name,
+            ", ".join([t.target_base for t in forbidden_targets])
+          ))
+
+    def _compute_fingerprint(self):
+      return stable_json_hash(self.banned_tag_name)
+
+  class TestDependencies(Constraint):
+    """Check that this target does not depend on test targets"""
+
+    def __init__(self):
+      self.name = "test_dependencies"
+
+    def _is_test_dependency(self, dependency):
+      return isinstance(dependency, JUnitTests)
+
+    def check_target(self, target, task_context, classes_relevant_to_target):
+      relevant_targets = DependencyConstraints.relevant_targets(target)
+      forbidden_targets = [t for t in relevant_targets if self._is_test_dependency(t)]
+      if forbidden_targets:
+        raise DependencyConstraint.BannedDependencyException(
+          'Target {} has test dependencies on targets ({})'.format(
+            target.target_base,
+            ", ".join([t.target_base for t in forbidden_targets])
+          ))
+
+    def _compute_fingerprint(self):
+      return stable_json_hash(self.name)
+
+
+class CheckBannedDeps(Task):
+  """
+  This task ensures that a target does not depend on banned dependencies.
+  """
+
+  @classmethod
+  def register_options(cls, register):
+    super(CheckBannedDeps, cls).register_options(register)
+    # If this flag changes, the debug message below should change too.
+    register('--enforce', type=bool, fingerprint=True,
```

Comment:
Please make this `skip` instead, since that is what is primarily used in the code base.